airbyte-cdk/bulk/toolkits/load-object-storage/src/main/kotlin/io/airbyte/cdk/load/file/object_storage/ObjectStoragePathFactory.kt

```diff
@@ -193,11 +193,15 @@ class ObjectStoragePathFactory(
                 PathVariable("EPOCH", """\d+""") { it.time.toEpochMilli().toString() },
                 PathVariable("UUID", """[a-fA-F0-9\\-]{36}""") { UUID.randomUUID().toString() }
             )
-        val PATH_VARIABLES_STREAM_CONSTANT = PATH_VARIABLES.filter { it.variable != "UUID" }
+        val PATH_VARIABLES_STREAM_CONSTANT =
+            PATH_VARIABLES.filter { it.variable == "NAMESPACE" || it.variable == "STREAM_NAME" }
         val FILENAME_VARIABLES =
             listOf(
                 FileVariable("date", """\d{4}_\d{2}_\d{2}""") { DATE_FORMATTER.format(it.time) },
-                FileVariable("timestamp", """\d+""") { it.time.toEpochMilli().toString() },
+                FileVariable("timestamp", """\d+""") {
+                    // NOTE: We use a constant time for the path but wall time for the files
+                    Instant.now().toEpochMilli().toString()
+                },
```

Comment:
Nit: Is there a way to use DI'd clock here? Also, this being in a companion object, when does it actually get called?

Reply:
If it's init, maybe we can initialize it in an init block and reference it

Reply:
The DI'd clock is "sync time". It's used once and the same value is constant across all uploads, and is used for all the time-based variables except this one, which is supposed to be the actual load time.